src/RefractAPI.cc

```diff
@@ -423,23 +488,29 @@ namespace drafter {
         return element;
     }
 
-    refract::IElement* CategoryToRefract(const snowcrash::Element& element)
+    refract::IElement* CategoryToRefract(const SectionInfo<snowcrash::Element>& element)
     {
         refract::ArrayElement* category = new refract::ArrayElement;
         RefractElements content;
 
         category->element(SerializeKey::Category);
 
-        if (element.category == snowcrash::Element::ResourceGroupCategory) {
+        if (element.section.category == snowcrash::Element::ResourceGroupCategory) {
             category->meta[SerializeKey::Classes] = CreateArrayElement(SerializeKey::ResourceGroup);
-            category->meta[SerializeKey::Title] = refract::IElement::Create(element.attributes.name);
+            category->meta[SerializeKey::Title] = PrimitiveToRefract(MAKE_SECTION_INFO(element, attributes.name));
         }
-        else if (element.category == snowcrash::Element::DataStructureGroupCategory) {
+        else if (element.section.category == snowcrash::Element::DataStructureGroupCategory) {
             category->meta[SerializeKey::Classes] = CreateArrayElement(SerializeKey::DataStructures);
         }
 
-        snowcrash::Elements elements = element.content.elements();
-        std::transform(elements.begin(), elements.end(), std::back_inserter(content), ElementToRefract);
+        if (!element.section.content.elements().empty()) {
+            const snowcrash::SourceMap<snowcrash::Elements>& sourceMap = element.sourceMap.content.elements().collection.empty()
+                ? SectionInfo<snowcrash::Elements>::NullSourceMap()
+                : element.sourceMap.content.elements();
```

Comment:
Let's use a proper `if...else` loop for this since this statement is spanning 3 lines anyway.